test/train/test_train_utils.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+from fruit_classifier.train.train_utils import get_image_paths
+from fruit_classifier.train.train_utils import get_data_and_labels
+from fruit_classifier.train.train_utils import get_model_input
+from fruit_classifier.train.train_utils import get_model
+from fruit_classifier.train.train_utils import train_model
+from fruit_classifier.train.train_utils import plot_training
+from fruit_classifier.preprocessing.preprocessing_utils import \
+    get_image_generator
+from pathlib import Path
+import shutil
+
+
+class TestTrainUtils(unittest.TestCase):
+    def setUp(self):
+        test_dir = Path(__file__).absolute().parents[1]
+
+        # Select a unique folder name for a new folder
+        folder_name = 'temp_folder'
+        self.folder_name = test_dir.joinpath(folder_name)
+        while self.folder_name.is_dir():
+            folder_name = folder_name + '_'
+            self.folder_name = test_dir.joinpath(folder_name)
+
+        # Make directory and two subdirectories
+        self.folder_name.mkdir(parents=True, exist_ok=True)
+        self.folder_name.joinpath('A').mkdir(parents=True, exist_ok=True)
+        self.folder_name.joinpath('B').mkdir(parents=True, exist_ok=True)
+
+        # Copy an image into each of those subdirectories
+        orig_file_path = test_dir.joinpath("test_data").\
+            joinpath("original_test_image.jpg")
+        shutil.copy(str(orig_file_path),
+                    str(self.folder_name.joinpath('A')))
+        shutil.copy(str(orig_file_path),
+                    str(self.folder_name.joinpath('B')))
+
+        # Run get_image_paths()
+        image_paths = get_image_paths(self.folder_name)
+        self.num_image_paths = len(image_paths)
+
+        # Run get_data_and_labels
+        data, labels = get_data_and_labels(image_paths)
+        self.num_data = len(data)
+        self.num_labels = len(labels)
+
+        # Run get_model_input
+        x_train, x_val, y_train, y_val = get_model_input(data, labels)
+        self.num_x_train = len(x_train)
+        self.num_x_val = len(x_val)
+        self.num_y_train = len(y_train)
+        self.num_y_val = len(y_val)
+
+        # Run get_image_generator
+        self.image_generator = get_image_generator()
+
+        # Run get_model
+        model = get_model(len(set(labels)), epochs=1)
+        self.model_was_built = model._built
+
+        # Run train_model
+        history = train_model(model,
```

Comment:
The `setUp` is for each member function of this class.
As there are 6 member functions you will train the same model with the same data and parameters 6 different times.
I would suggest to move the training to `test_train_model`

Reply:
But I will not be able to let the output from test_train_model(), and use it in test_outputs_were_right()? In that case I should also verify the outputs (the 6 tests) inside of test_train_model()?

Reply:
I have now gathered the entire training, aswell as the testing of each output into one function.